examples/nidigital_spi/measurement.py

```diff
@@ -159,6 +158,12 @@ def _resolve_relative_path(
         return (directory_path / file_path).resolve()
 
 
+def _get_file_name_from_path(file_path: Union[str, pathlib.Path]) -> str:
+    if not isinstance(file_path, pathlib.Path):
+        file_path = pathlib.Path(file_path)
```

Comment:
hmmmm...

normally, you would just do something like
```python
pathlib_file_path = pathlib.Path(file_path)
return pathlib_file_path.stem
```

however, it is fair to say that always copying is likely (I haven't tested it) not as performant as an if-type.

Reply:
I'm not python-enlightened enough to have an opinion, but I was copying the code pattern from lines 153-154, so I'm guessing this comment applies to that too?

I'm not sure why any of the usages of either function would ever be passing in a pathlib.Path based on my reading of the code. @bkeryan are these ever going to be pathlib.Paths?

Either way, does it maybe make sense to do the pathlib.Path() creation in the measure function before calling either of these, and simplifying these to expect a pathlib.Path always?

Reply:
Matthew's suggestion is fine. The overhead of constructing an extra `pathlib.Path` is not important, so remove the type check.

Also, since this function doesn't access the filesystem, `pathlib.PurePath` is fine.

> I'm not sure why any of the usages of either function would ever be passing in a pathlib.Path based on my reading of the code. @bkeryan are these ever going to be pathlib.Paths?

I would prefer to represent paths as `pathlib.Path` or `pathlib.PurePath`, and I originally tried to use it in this example's `measure` function, but NI's Python APIs do not support `pathlib` yet so I switched to `str` before I posted the PR to create this example.

https://github.com/ni/measurementlink-python/issues/283
https://github.com/ni/nidaqmx-python/issues/26
https://github.com/ni/nimi-python/issues/1949

Reply:
> pathlib_file_path = pathlib.Path(file_path)

FYI, I think mypy allows reusing a variable name for type narrowing.

![image](https://github.com/ni/measurementlink-python/assets/17914399/fb9f5474-222a-4a26-a2cc-9f46974b19cf)

Reply:
it does -> I tend to use a new name to and try to communicate to the next reader (myself next week, or team-mate next month) why I'm doing a type conversion. (not saying this name is the best, though)